File: templates/workspace-dialect/hello.ts
import { op, OpContext, Value } from "forge";

interface Greeting {
  message: string;
}

export const hello = op("hello", (ctx: OpContext, input: Value) => {
  const name: string = input.get("name", "world");
  const out: Greeting = { message: "hello, " + name + ", from {{domain}}" };
  return out;
});

File: templates/workspace-dialect/hello.py
from dataclasses import dataclass

from forge import OpContext, Value, op


@dataclass
class Greeting:
    message: str


@op("hello")
def hello(ctx: OpContext, input: Value) -> Greeting:
    name: str = input.get("name", "world")
    return Greeting(message="hello, " + name + ", from {{domain}}")
